modules/azure/logic_app_standard/main.tf

```
@@ -54,11 +58,26 @@ resource "azurerm_logic_app_standard" "app" {
   virtual_network_subnet_id  = var.integration_subnet_id
 }
 
-# First, create a zip file containing the workflow
-data "archive_file" "workflow" {
+# First, create a check.zip with archive_file to check diffs (this step is required)
+# replacing this step by checking of deploy.zip created by local-exec doesn't work
+# because local-exec is not executed during 'plan' so it would take old deploy.zip
+data "archive_file" "check_zip" {
   type        = "zip"
   source_dir  = var.workflows_source_path
-  output_path = "${path.module}/files/deploy.zip"
+  output_path = "${path.module}/files/check.zip"
+}
+
+resource "null_resource" "zip_logic_app" {
+  depends_on = [data.archive_file.check_zip]
+
+  triggers = {
+    deploy = data.archive_file.check_zip.output_sha
+  }
+  # if check.zip file changes, create deploy.zip file
+  provisioner "local-exec" {
+    interpreter = local.is_linux ? ["bash", "-c"] : ["PowerShell", "-Command"]
+    command     = local.is_linux ? "cd ${path.module} && mkdir -p files && cd files && cd ${var.workflows_source_path} && zip -rq $OLDPWD/deploy.zip ." : "New-Item -Path \"${path.module}\" -Name \"files\" -ItemType \"directory\" -Force; Compress-Archive -Path \"${var.workflows_source_path}\\*\" -DestinationPath \"${path.module}\\files\\deploy.zip\""
```

Would not one of these more readable?

```suggestion
    command     = local.is_linux ? "cd ${path.module} && mkdir -p files && zip -rq files/deploy.zip ${var.workflows_source_path}" : "New-Item -Path \"${path.module}\" -Name \"files\" -ItemType \"directory\" -Force; Compress-Archive -Path \"${var.workflows_source_path}\\*\" -DestinationPath \"${path.module}\\files\\deploy.zip\""
```

```suggestion
    command     = local.is_linux ? "cd ${path.module} && mkdir -p files && zip -rq ${path.module}/files/deploy.zip ${var.workflows_source_path}" : "New-Item -Path \"${path.module}\" -Name \"files\" -ItemType \"directory\" -Force; Compress-Archive -Path \"${var.workflows_source_path}\\*\" -DestinationPath \"${path.module}\\files\\deploy.zip\""
```

---

Would this
```suggestion
    command     = local.is_linux ? "cd ${path.module} && mkdir -p files && cd ${var.workflows_source_path} && zip -rq $OLDPWD/files/deploy.zip ." : "New-Item -Path \"${path.module}\" -Name \"files\" -ItemType \"directory\" -Force; Compress-Archive -Path \"${var.workflows_source_path}\\*\" -DestinationPath \"${path.module}\\files\\deploy.zip\""
```
or this
```suggestion
    command     = local.is_linux ? "cd ${path.module} && mkdir -p files && cd ${var.workflows_source_path} && zip -rq ${path.module}/files/deploy.zip ." : "New-Item -Path \"${path.module}\" -Name \"files\" -ItemType \"directory\" -Force; Compress-Archive -Path \"${var.workflows_source_path}\\*\" -DestinationPath \"${path.module}\\files\\deploy.zip\""
```
work?

---

The first one works, the second one throws an error.